src/twisted/python/compat.py

```diff
@@ -377,13 +399,17 @@ def _matchingString(constantString, inputString):
     @rtype: the type of C{inputString}
     """
     if isinstance(constantString, bytes):
-        otherType = constantString.decode("ascii")
-    else:
-        otherType = constantString.encode("ascii")
-    if type(constantString) == type(inputString):
-        return constantString
+        uniString = constantString.decode("ascii")
```

Comment:
Why decode here instead of in the `else` clause? Seems like it might get computed here and then thrown away.

Reply:
I mentioned this in the commit message, but: the function is explicitly documented as encoding or decoding its argument even i it doesn't have to. I'd rather not break that stated contract, even if it is a silly one.